src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -37,6 +37,12 @@ internal abstract class AbstractPullDiagnosticHandler<TDiagnosticsParams, TRepor
         /// </summary>
         private static readonly Option2<DiagnosticMode> s_liveShareDiagnosticMode = new(nameof(InternalDiagnosticsOptions), "LiveShareDiagnosticMode", defaultValue: DiagnosticMode.Pull);
 
+        /// <summary>
+        /// Diagnostic mode setting for .editorconfig files.  This should always be <see cref="DiagnosticMode.Pull"/> as there is no push support in .editorconfig files.
+        /// This option is only for passing to the diagnostics service and can be removed when we switch all of Roslyn to LSP pull.
+        /// </summary>
+        private static readonly Option2<DiagnosticMode> s_editorConfigDiagnosticMode = new(nameof(InternalDiagnosticsOptions), "EditorConfigDiagnosticMode", defaultValue: DiagnosticMode.Pull);
```

Comment:
I think this can be removed - instead we'll just support editorconfig diagnostics only when the normal diagnostic mode is in LSP.